datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -48,6 +48,15 @@ def __contains__(self, key):
     def get(self, key):
         return super(CaseInsensitiveDict, self).get(key.lower())
 
+    def copy(self):
+        """
+        Explicit copy to ensure keys are consistent
+        """
+        copied = CaseInsensitiveDict()
+        for key, value in self.items():
+            copied[key] = value
+        return copied
```

Comment:
```suggestion
        return CaseInsensitiveDict(self)
```
?

Reply:
It's probably unlikely to happen but I used the test event in `FakeWMISampler` that wasn't case insensitive and casting `CaseInsensitiveDict` on the event would result in keyError with `__get__` calls. I thought an explicit copy would be safer. Would it be better to simplify since the above case is unlikely as the sampler creates CaseInsensitiveDict events?

Reply:
I think it would be better to fix `FakeWMISampler` to return a `CaseInsensitiveDict` to return the correct type. WDYT?

Reply:
That works too! I've updated the the example event to reflect CaseInsensitiveDict type.